internnav/agent/dialog_agent.py

```diff
@@ -0,0 +1,468 @@
+import argparse
+import copy
+import itertools
+import random
+import re
+import time
+from collections import OrderedDict
+from typing import Any, Dict
+
+import numpy as np
+import quaternion
+import torch
+from PIL import Image, ImageDraw
+from transformers import (
+    AutoProcessor,
+    AutoTokenizer,
+    Qwen2_5_VLForConditionalGeneration,
+)
+
+from internnav.agent import Agent
+from internnav.configs.agent import AgentCfg
+
+try:
+    pass
+except Exception as e:
+    print(f"Warning: ({e}), Ignore this if not using dual_system.")
+
+try:
+    from depth_camera_filtering import filter_depth
+    from habitat.tasks.nav.shortest_path_follower import ShortestPathFollower
+except Exception as e:
+    print(f"Warning: ({e}), Habitat Evaluation is not loaded in this runtime. Ignore this if not using Habitat.")
+
+DEFAULT_IMAGE_TOKEN = "<image>"
+
+
+def split_and_clean(text):
+    import re
+
+    parts = re.split(r'(<image>)', text)
+    results = []
+    for part in parts:
+        if part == '<image>':
+            results.append(part)
+        else:
+            clean_part = part.replace('\n', '').strip()
+            if clean_part:
+                results.append(clean_part)
+    return results
+
+
+@Agent.register('dialog')
+class DialogAgent(Agent):
+    """
+    agent template, override the functions for custom policy
```

Comment:
Can we have a more formal docstring for this class, which I think is important as a base class for tasks involving dialogs?